src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -282,7 +282,7 @@ export class ConnectionWidget extends lifecycle.Disposable {
 	private addDatabaseOption(): void {
 		// Database
 		let databaseOption = this._optionsMaps[ConnectionOptionSpecialType.databaseName];
-		if (databaseOption) {
+		if (databaseOption && !databaseOption.hide) {
```

Comment:
Is this fixing a bug or something?

Reply:
I added this flag because ADS requires a Database to be declared in the package.json for the Query window to work. This causes a database field to be rendered in the Connection Dialog. I added this flag to allow me to declare it while not rendering it.

Reply:
Username / Password is required to be declared in the Package.json even if the connection type doesn't require it.

Reply:
Hm, could you submit this as a separate PR? This seems more like a workaround then an actual fix and I'd like us to review it separately so we can try to make this stuff easier to use. 

And regardless I believe that you should be adding this new property to the ConnectionOption interface https://github.com/microsoft/azuredatastudio/blob/main/src/sql/azdata.d.ts#L582. Although this logic seems weird to me to begin with since optionsMap says the keys should be numbers but we're ever only using strings...so not sure what's going on.